psi4/driver/procrouting/dft_funcs/dict_builder.py

```diff
@@ -0,0 +1,347 @@
+       "X_METHOD_NAME":  {       must match a LibXC method  
+                 "alpha": 1.0,   coefficient for (global) GGA exchange, by default 1.0
+                 "omega": 0.0,   range-separation parameter
+             "use_libxc": False  whether "x_hf" parameters should be set from LibXC values for this method
+                 "tweak": [],    tweak the underlying functional
+     },
+      
+           "x_hf":  {          definition of HF exchange for hybrid functionals
+               "alpha": 0.0,             coefficient for (global) HF exchange, by default none
+                "beta": 0.0,             coefficient for long range HF exchange
+               "omega": 0.0,             range separation parameters
+           "use_libxc": "X_METHOD_NAME"  reads the above 3 values from specified X functional
+     },
+           
+  "c_functionals":  {          definition of C contributions
+       "C_METHOD_NAME":  {       must match a LibXC method  
+                 "alpha": 1.0,   coefficient for (global) GGA correlation, by default 1.0
+                 "tweak": [],    tweak the underlying functional
+    },   
+   
+          "c_mp2":  {          definition of MP2 correlation double hybrid functionals
+               "alpha": 0.0,     coefficient for MP2 correlation, by default none
+                  "ss": 0.0,     coefficient for same spin correlation in SCS methods, forces alpha = 1.0
+                  "os": 0.0,     coefficient for opposite spin correlation in SCS methods, forces alpha = 1.0
+    },
+          
+     "dispersion":  {          definition of dispersion corrections
+               "type": "",       dispersion type - "d2", "d3zero", "d3bj" etc., see empirical_dispersion.py
+             "params": {},       parameters for the dispersion correction
+           "citation": "",       special reference for the dispersion correction, appended to output
+    },
+}
+"""
+from psi4 import core
+from psi4.driver.p4util.exceptions import *
+from psi4.driver.qcdb.dashparam import dashcoeff
+
+import copy
+
+from . import dict_xc_funcs
+from . import dict_lda_funcs
+from . import dict_gga_funcs
+from . import dict_mgga_funcs
+from . import dict_hyb_funcs
+from . import dict_dh_funcs
+
+dict_functionals = {}
+dict_functionals.update(dict_xc_funcs.functional_list)
+dict_functionals.update(dict_lda_funcs.functional_list)
+dict_functionals.update(dict_gga_funcs.functional_list)
+dict_functionals.update(dict_mgga_funcs.functional_list)
+dict_functionals.update(dict_hyb_funcs.functional_list)
+dict_functionals.update(dict_dh_funcs.functional_list)
+
+
+def get_functional_aliases(functional_dict):
+    if "alias" in functional_dict.keys():
+        aliases = [each.upper() for each in functional_dict["alias"]]
+        aliases.append(functional_dict["name"].upper())
+    else:
+        aliases = [functional_dict["name"].upper()]
+    return aliases
+
+
+# alias table for dispersion
+dispersion_names = {}
```

Comment:
We should probably move this to EmpiricalDispersion (or perhaps dashparams), so that when new corrections are added (VV10), it doesn't need to be edited in 20 places.

Reply:
+1 for that.

Reply:
I've moved it to `EmpiricalDispersion`. I don't like how "messy" the current interface is, scattered over a few files. Ideally, we'd load the `dashparams` from a `.json` file, and the `EmpiricalDispersion` module would be self-contained. I might make a pull request with some ideas once this is one is through... It would be good to get input from @hokru 's work on VV10 so that we're all on the same page.